modules/imap/handler_modules.php

```diff
@@ -907,6 +907,86 @@ public function process() {
     }
 }
 
+/**
+ * Snooze message
+ * @subpackage imap/handler
+ */
+class Hm_Handler_imap_snooze_message extends Hm_Handler_Module {
+    /**
+     * Use IMAP to snooze the selected message uid
+     */
+    public function process() {
+        list($success, $form) = $this->process_form(array('imap_snooze_ids', 'imap_snooze_until'));
+        if (!$success) {
+            return;
+        }
+        $snoozed_messages = 0;
+        $snooze_tag = null;
+        if ($form['imap_snooze_until'] != 'unsnooze') {
+            $at = date('D, d M Y H:i:s O');
+            $until = get_snooze_date($form['imap_snooze_until']);
+            $snooze_tag = "X-Snoozed: at $at;\n \tuntil $until";
+        }
+        $ids = explode(',', $form['imap_snooze_ids']);
+        foreach ($ids as $msg_part) {
+            list($imap_server_id, $msg_id, $folder) = explode('_', $msg_part);
+            $cache = Hm_IMAP_List::get_cache($this->cache, $imap_server_id);
+            $imap = Hm_IMAP_List::connect($imap_server_id, $cache);
+            if (imap_authed($imap)) {
+                $folder = hex2bin($folder);
+                if (snooze_message($imap, $msg_id, $folder, $snooze_tag)) {
+                    $snoozed_messages++;
+                }
+            }
+        }
+        $this->out('snoozed_messages', $snoozed_messages);
+        if ($snoozed_messages == count($ids)) {
+            $msg = 'Messages snoozed';
+        } elseif ($snoozed_messages > 0) {
+            $msg = 'Some messages have been snoozed';
+        } else {
+            $msg = 'ERRFailed to snooze selected messages';
```

Comment:
`ERRFailed...` -> `ERR Failed...`

Reply:
It is actually correct without a space. That's how it is done throughout Cypht and Hm_Output_msgs strips the first 3 chars in case they are 'ERR' to show a formatted error message. Adding a space will add it to the message when displaying in the frontend.